### PR TITLE
Migrate to JUnit 5

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR

Legacy JUnit 4 was used

## After this PR
==COMMIT_MSG==
Migrates all tests to JUnit 5 Jupiter
==COMMIT_MSG==

### CODE REVIEW

File: tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistryTest.java [hunk 1]
```diff
@@ -96,18 +74,21 @@ public void before() {
         Mockito.verify(mockSupplier, times(2)).get();
     }
 
-    @Test
-    public void testCounter() {
+    @ParameterizedTest
+    @MethodSource("com.palantir.tritium.registry.test.TestTaggedMetricRegistries#registries")
```
Comment (carterkozak) [inline tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistryTest.java]: Thoughts on extracting `"com.palantir.tritium.registry.test.TestTaggedMetricRegistries#registries"` to a constant in `TestTaggedMetricRegistries`?

Comment (schlosna) [reply tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistryTest.java]: Yeah, can extract these to constants. I can probably tweak the projects’ test dependencies to use a single implementation as well.

File: tritium-metrics/build.gradle [hunk 2]
```diff
@@ -23,13 +23,14 @@ dependencies {
 
     testImplementation project(':tritium-test')
```
Comment (schlosna) [inline tritium-metrics/build.gradle:24]: See if we can share a single `TestTaggedMetricRegistries`

```suggestion
    testImplementation project(':tritium-registry')
    testImplementation project(':tritium-test')
```

Comment (schlosna) [reply tritium-metrics/build.gradle:24]: skipping as this ends up being nasty with gradle test dependencies, just going to leave the duplication for now

**Comment by carterkozak** [comment]
lgtm :+1:

**Pull request merged**